src/common/com/intellij/plugins/haxe/lang/psi/impl/HaxeReferenceImpl.java

```diff
@@ -740,11 +740,18 @@ private static void addClassStaticMembersVariants(Set<HaxeComponentName> suggest
 
     boolean extern = haxeClass.isExtern();
     boolean isEnum = haxeClass instanceof HaxeEnumDeclaration;
+    boolean isAbstractEnum = HaxeAbstractEnumUtil.isAbstractEnum(haxeClass);
 
     for (HaxeNamedComponent namedComponent : HaxeResolveUtil.findNamedSubComponents(haxeClass)) {
       final boolean needFilter = filterByAccess && !namedComponent.isPublic();
-      if ((extern || !needFilter) && (namedComponent.isStatic() || isEnum) && namedComponent.getComponentName() != null) {
-        suggestedVariants.add(namedComponent.getComponentName());
+      final HaxeComponentName componentName = namedComponent.getComponentName();
+      if(componentName != null) {
+        if(isAbstractEnum && HaxeAbstractEnumUtil.isAbstractEnumField(namedComponent)) {
+          suggestedVariants.add(componentName);
```

Comment:
HaxeAbstractEnumUtil.isAbstractEnumField - checks each time if class of namedComponent is AbstractClass, maybe something can be simplified, but mostly ok